src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -304,6 +304,13 @@ private BuildEventArgs ReadTargetSkippedEventArgs()
                 condition = ReadOptionalString();
                 evaluatedCondition = ReadOptionalString();
                 originallySucceeded = ReadBoolean();
+
+                // Attempt to infer skip reason from the data we have
+                skipReason = condition != null ?
```

Comment:
This is saying that if the condition was not null, the condition was false, which doesn't sound right.

Reply:
Yes, if the target was skipped because of the false condition, we store the condition which was false. If it was skipped for other reasons, the condition will be null.

Reply:
And if the condition was true, but we skipped it for another reason, condition would still be null instead of true? That's confusing, but ok.

Reply:
Yes, we only store the Condition if the target was skipped because of false condition ;)